design/ceph/ceph-telemetry.md

```diff
@@ -0,0 +1,77 @@
+# Adding Rook data to Ceph telemetry
+The solution plan agreed upon with the telemetry team is for the Rook operator to add telemetry to
+the Ceph mon `config-key` database, and Ceph will read each of those items for telemetry retrieval.
+See: https://github.com/rook/rook/pull/10161
+
+## Guidelines from the Ceph telemetry team
+- Users must opt in to telemetry
+- Users must also opt into the new (delta) telemetry items added between Ceph versions
+- Rook should not add information to `config-key` keys that can grow arbitrarily large to keep space
+  usage of the mon database low.
+
+## Proposed metrics for Rook to collect
+- `rook-version` - Rook version.
+  - Done in https://github.com/rook/rook/pull/10161
+- `rook-kubernetes-version` - Kubernetes version.
+  - This is similar to the already collected os/kernel versions.
+- `rook-num-kubernetes-nodes` - Number of Kubernetes nodes.
+- `rook-ceph-csi-version` - Ceph CSI version.
+- Num CSI plugin pods running - This gives idea of how many k8s nodes are in the k8s cluster that
+  are allowed to use Ceph storage.
+  - It is easiest count plugin pods separately since each DaemonSet is a different label. Plus, it
+    can show if some CSI drivers are ever disabled.
+  - `rook-num-csi-rbd-plugin-pods`
+  - `rook-num-csi-cephfs-plugin-pods`
+  - `rook-num-csi-nfs-plugin-pods`
+- `rook-num-nodes` - Number of nodes on which the cluster has running pods.
+  - Since clusters with portable PVCs have one "node" per PVC, this will help show the actual node
+    count for Rook installs in portable environments
+  - We can get this by counting the number of crash collector pods; however, if any users disable
+    the crash collector, we should report an unknown value, possibly `unknown`.
+- `rook-num-storage-classes` - Number of storage classes related to the current Ceph cluster. We can
+  also break this up by specific kinds of storage classes, but the overall may be useful if new
+  classes are added in the future.
+  - `rook-num-rbd-storage-classes`
+  - `rook-num-cephfs-storage-classes`
+  - `rook-num-nfs-storage-classes`
+  - `rook-num-bucket-storage-classes`
+- `rook-num-storage-class-device-sets` - Number of storage class device sets.
+- `rook-num-portable-storage-class-device-sets` - .
+- `rook-mons-are-pvc` - true/false whether mons are on PVC.
+- `rook-network-provider` - The network provider used for the cluster. e.g., default, host, multus.
+- `rook-external-mode` - true/false if the cluster is in external mode.
+
```

Comment:
Perhaps we could have sections of metrics in this doc so we can see what's related to the cluster CR, or overall metrics, etc? Then it may be easier to see if metrics are needed for particular CRs, and so on.

Reply:
Should we add all consts at once, or only when implemented?

Reply:
I would expect the consts would only be added when implemented.